services/audit-service/src/main/java/datawave/microservice/audit/auditors/file/FileAuditor.java

```diff
@@ -0,0 +1,220 @@
+package datawave.microservice.audit.auditors.file;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import datawave.webservice.common.audit.AuditParameters;
+import datawave.webservice.common.audit.Auditor;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.LocalFileSystem;
+import org.apache.hadoop.fs.Path;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URLEncoder;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * An implementation for {@link Auditor}, which writes URL encoded, JSON formatted audit messages to a file.
+ */
+public class FileAuditor implements Auditor {
+    
+    private final Logger log = LoggerFactory.getLogger(this.getClass());
+    
+    protected static final ObjectMapper mapper = new ObjectMapper();
+    
+    protected final SimpleDateFormat sdf;
+    protected final ReentrantLock writeLock = new ReentrantLock(true);
+    
+    protected long maxFileLenBytes;
+    protected long maxFileAgeMillis;
+    
+    protected FileSystem fileSystem;
```

Comment:
Note that this is a Closeable object.  If this will exist for the lifetime of the jvm, then no problem.  Otherwise we need a close/cleanup mechanism.  A configuration refresh may be a path of concern here.

Reply:
It will live the lifetime of the container, so I'll leave it alone for now.